code/modules/logging/log_holder.dm

```diff
@@ -206,13 +206,26 @@ GENERAL_PROTECT_DATUM(/datum/log_holder)
 
 	return category_tree
 
+#define UPDATE_NEW_CATEGORY_INSTANCE_FROM_OLD_IF_NEEDED(instance_ident, log_error) \
+	if(##instance_ident.category in log_categories) { \
+		var/datum/log_category/sacrifical_pawn = log_categories[##instance_ident.category]; \
+		##instance_ident.entry_count = sacrifical_pawn.entry_count; \
+		##instance_ident.entries = sacrifical_pawn.entries; \
+		if(UNLINT(log_error)) { \
+			Log(LOG_CATEGORY_INTERNAL_ERROR, "re-initializing a log category that has already been initialized.", list("category" = ##instance_ident.category)); \
+		} \
```

Comment:
This branch should be something you call yourself in the caller for the proc this will become. As in...
```dm
if (!whatever(src))
    Log(etc)
```